runtime/interpreter/value.go

```diff
@@ -20601,6 +20601,13 @@ func (v *StorageReferenceValue) dereference(interpreter *Interpreter, locationRa
 		return nil, nil
 	}
 
+	if reference, isReference := referenced.(ReferenceValue); isReference {
+		panic(NestedReferenceError{
+			Value:         reference,
+			LocationRange: locationRange,
+		})
+	}
```

Comment:
Can we do this check at the time of the creation? i.e: in the constructor `NewUnmeteredStorageReferenceValue`

Reply:
ah never mind, just saw the reason for doing it here.